selfservice/flow/recovery/flow.go

```diff
@@ -130,7 +130,12 @@ func NewFlow(conf *config.Config, exp time.Duration, csrf string, r *http.Reques
 }
 
 func FromOldFlow(conf *config.Config, exp time.Duration, csrf string, r *http.Request, strategies Strategies, of Flow) (*Flow, error) {
-	nf, err := NewFlow(conf, exp, csrf, r, strategies, of.Type)
+	f := of.Type
```

Comment:
Is this method only called for expired flows, or could it break some other behavior?

Reply:
If i understand [this code](https://github.com/ory/kratos/blob/master/selfservice/flow/verification/error.go#L70) correctly, then yes, only if we get an `ExpiredError`

Reply:
It appears that the code is a bit messy, for some of the flows we set the flow type to browser:

https://github.com/ory/kratos/blob/c5a04b5f174e06faca99ebc7461c8ebe8e1f694d/selfservice/strategy/link/strategy_recovery.go#L310

for others, we seem to reuse the flow type:

https://github.com/ory/kratos/blob/c5a04b5f174e06faca99ebc7461c8ebe8e1f694d/selfservice/strategy/link/strategy_recovery.go#L244

so we need to make sure that this PR here does not add any unwanted regressions :/

Reply:
If this exists only on verification, i can remove the added logic from recovery flow, that should ensure we do not introduce any regressions there

Reply:
So I checked, this is only called on expired flows.